test/e2e-multicluster-test/multicluster_test.go

```diff
@@ -501,5 +501,59 @@ var _ = Describe("Test multicluster scenario", func() {
+							Type:       "k8s-objects",
+							Properties: &runtime.RawExtension{Raw: []byte(`{"objects":[{"apiVersion":"v1","kind":"ConfigMap","metadata":{"name":"non-shared-` + name + `"},"data":{"key":"value"}}]}`)},
+						}},
+						Policies: []v1beta1.AppPolicy{{
+							Type:       "shared-resource",
+							Name:       "shared-resource",
+							Properties: &runtime.RawExtension{Raw: []byte(`{"rules":[{"selector":{"CompNames":["shared-resource-` + name + `"]}}]}`)},
+						}},
+					},
+				}
+			}
+			app1 := createApp("app1")
+			Expect(k8sClient.Create(hubCtx, app1)).Should(Succeed())
+			app2 := createApp("app2")
+			Expect(k8sClient.Create(hubCtx, app2)).Should(Succeed())
+			app3 := createApp("app3")
+			Expect(k8sClient.Create(hubCtx, app3)).Should(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "shared"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app1"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app2"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app3"}, &corev1.ConfigMap{})).Should(Succeed())
+			}, 45*time.Second).Should(Succeed())
+			Expect(k8sClient.Delete(hubCtx, app2)).Should(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "shared"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app1"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app2"}, &corev1.ConfigMap{})).Should(Satisfy(kerrors.IsNotFound))
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app3"}, &corev1.ConfigMap{})).Should(Succeed())
+			}, 10*time.Second).Should(Succeed())
+			Expect(k8sClient.Delete(hubCtx, app1)).Should(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "shared"}, &corev1.ConfigMap{})).Should(Succeed())
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app1"}, &corev1.ConfigMap{})).Should(Satisfy(kerrors.IsNotFound))
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "non-shared-app3"}, &corev1.ConfigMap{})).Should(Succeed())
+			}, 10*time.Second).Should(Succeed())
+			Expect(k8sClient.Delete(hubCtx, app3)).Should(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(hubCtx, types.NamespacedName{Namespace: namespace, Name: "shared"}, &corev1.ConfigMap{})).Should(Satisfy(kerrors.IsNotFound))
```

Comment:
you should check the annotation of the shared resource, check if it matches the expected state.